### ISSUE TITLE
NotImplementedError: Redirects not yet supported

### ISSUE THREAD

**Issue by mr-engin3er** [rdehuyss/micropython-ota-updater#20]
Board = ESP32
Micropython Version = v1.15
Github repo type = Private
Authorization = Personal Access Token
i print l in the request method of HTTP_CLIENT and got logs from HTTP_CLIENT

`l:  b'HTTP/1.1 200 OK\r\n'
l:  b'Server: GitHub.com\r\n'
l:  b'Date: Fri, 07 May 2021 06:39:37 GMT\r\n'
l:  b'Content-Type: application/json; charset=utf-8\r\n'
l:  b'Content-Length: 1794\r\n'
l:  b'Cache-Control: private, max-age=60, s-maxage=60\r\n'
l:  b'Vary: Accept, Authorization, Cookie, X-GitHub-OTP\r\n'
l:  b'ETag: "363008357f30821d99f3d17e9391d5ce15b8d7334be544a0f7d36831f4bfc72f"\r\n'
l:  b'Last-Modified: Thu, 06 May 2021 13:05:17 GMT\r\n'
l:  b'X-OAuth-Scopes: repo\r\n'
l:  b'X-Accepted-OAuth-Scopes: repo\r\n'
l:  b'X-GitHub-Media-Type: github.v3; format=json\r\n'
l:  b'X-RateLimit-Limit: 5000\r\n'
l:  b'X-RateLimit-Remaining: 4999\r\n'
l:  b'X-RateLimit-Reset: 1620373177\r\n'
l:  b'X-RateLimit-Used: 1\r\n'
l:  b'X-RateLimit-Resource: core\r\n'
l:  b'Access-Control-Expose-Headers: ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Used, X-RateLimit-Resource, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type, Deprecation, Sunset\r\n'
l:  b'Access-Control-Allow-Origin: *\r\n'
l:  b'Strict-Transport-Security: max-age=31536000; includeSubdomains; preload\r\n'
l:  b'X-Frame-Options: deny\r\n'
l:  b'X-Content-Type-Options: nosniff\r\n'
l:  b'X-XSS-Protection: 0\r\n'
l:  b'Referrer-Policy: origin-when-cross-origin, strict-origin-when-cross-origin\r\n'
l:  b"Content-Security-Policy: default-src 'none'\r\n"
l:  b'Vary: Accept-Encoding, Accept, X-Requested-With\r\n'
l:  b'X-GitHub-Request-Id: 4AC8:439E:B9DA6:E015A:6094E0A6\r\n'
l:  b'connection: close\r\n'
Checking version... 
        Current version:  0.0
        Latest version:  v0.0.2
New version available, will download and install on next reboot
New update found:  v0.0.2
network config: ('192.168.***.***', '255.255.255.0', '192.168.***.***', '192.168.***.***')
l:  b'HTTP/1.1 200 OK\r\n'
l:  b'Server: GitHub.com\r\n'
l:  b'Date: Fri, 07 May 2021 06:39:41 GMT\r\n'
l:  b'Content-Type: application/json; charset=utf-8\r\n'
l:  b'Content-Length: 1794\r\n'
l:  b'Cache-Control: private, max-age=60, s-maxage=60\r\n'
l:  b'Vary: Accept, Authorization, Cookie, X-GitHub-OTP\r\n'
l:  b'ETag: "363008357f30821d99f3d17e9391d5ce15b8d7334be544a0f7d36831f4bfc72f"\r\n'
l:  b'Last-Modified: Thu, 06 May 2021 13:05:17 GMT\r\n'
l:  b'X-OAuth-Scopes: repo\r\n'
l:  b'X-Accepted-OAuth-Scopes: repo\r\n'
l:  b'X-GitHub-Media-Type: github.v3; format=json\r\n'
l:  b'X-RateLimit-Limit: 5000\r\n'
l:  b'X-RateLimit-Remaining: 4998\r\n'
l:  b'X-RateLimit-Reset: 1620373177\r\n'
l:  b'X-RateLimit-Used: 2\r\n'
l:  b'X-RateLimit-Resource: core\r\n'
l:  b'Access-Control-Expose-Headers: ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Used, X-RateLimit-Resource, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type, Deprecation, Sunset\r\n'
l:  b'Access-Control-Allow-Origin: *\r\n'
l:  b'Strict-Transport-Security: max-age=31536000; includeSubdomains; preload\r\n'
l:  b'X-Frame-Options: deny\r\n'
l:  b'X-Content-Type-Options: nosniff\r\n'
l:  b'X-XSS-Protection: 0\r\n'
l:  b'Referrer-Policy: origin-when-cross-origin, strict-origin-when-cross-origin\r\n'
l:  b"Content-Security-Policy: default-src 'none'\r\n"
l:  b'Vary: Accept-Encoding, Accept, X-Requested-With\r\n'
l:  b'X-GitHub-Request-Id: 4ACD:6C73:35B41:5B3BE:6094E0AA\r\n'
l:  b'connection: close\r\n'
Checking version... 
        Current version:  0.0
        Latest version:  v0.0.2
Updating to version v0.0.2...
Downloading version v0.0.2
url:  https://api.github.com/repos/user/private-repo/contents/src/?ref=refs/tags/v0.0.2
l:  b'HTTP/1.1 302 Found\r\n'
l:  b'Server: GitHub.com\r\n'
l:  b'Date: Fri, 07 May 2021 06:39:45 GMT\r\n'
l:  b'Content-Type: text/html;charset=utf-8\r\n'
l:  b'Content-Length: 0\r\n'
l:  b'Location: https://api.github.com/repositories/264506430/contents/src\r\n'`

Error logs
`Traceback (most recent call last):
  File "main.py", line 51, in <module>
  File "main.py", line 46, in boot
  File "main.py", line 28, in download_and_install_update_if_available
  File "app/ota_updater.py", line 59, in install_update_if_available_after_boot
  File "app/ota_updater.py", line 81, in install_update_if_available
  File "app/ota_updater.py", line 132, in _download_new_version
  File "app/ota_updater.py", line 139, in _download_all_files
  File "app/github_http_client.py", line 145, in get
  File "app/github_http_client.py", line 131, in request
NotImplementedError: Redirects not yet supported
MicroPython v1.15 on 2021-04-18; ESP32 module with ESP32`

I hope it'll be added soon in the future but I need it on a project so if anyone can guide me on how to achieve this so I can customize  HTTP_CLIENT code and if I achieve I'll post it here.

**Comment by mr-engin3er** [comment]
some helpful resources for redirect implementation:- 
1.  https://github.com/micropython/micropython-lib/blob/master/urequests/urequests.py
2.  https://github.com/micropython/micropython-lib/pull/398 pull request.
3.  https://github.com/micropython/micropython-lib/pull/378 pull request.
4.  https://github.com/micropython/micropython-lib/pull/368 pull request.

**Comment by mr-engin3er** [comment]
Finally, Somehow I implement redirect feature.
I'm not an expert, but I saw some pull request from official microython urequest library 
and made some changes on microopython-ota-updater library and created a pull request #21
I personally tested this codes on my esp32 and micropython v1.15.
If someone wants to try they can use my [repo](https://github.com/mr-engin3er/micropython-ota-updater) which was forked of this library. 
This is my first contribution to open source so if I made any mistake then let me correct.
